fili-core/src/main/java/com/yahoo/bard/webservice/table/BasePhysicalTable.java

```diff
@@ -139,4 +116,49 @@ public String getPhysicalColumnName(String logicalName) {
     protected void setAvailability(Availability availability) {
         this.availability = availability;
     }
+
+    /**
+     * Create a constrained copy of this table.
+     *
+     * @param constraint  The dataSourceConstraint which narrows the view of the underlying availability
+     *
+     * @return a constrained table whose availability and serialization are narrowed by this constraint
+     */
+    public ConstrainedTable withConstraint(DataSourceConstraint constraint) {
+        validateConstraintSchema(constraint);
+        return new ConstrainedTable(this, new PhysicalDataSourceConstraint(constraint, getSchema()));
+    }
+
+    @Override
+    public Set<TableName> getDataSourceNames() {
+        return getAvailability().getDataSourceNames();
+    }
+
+    /**
+     * Ensure that the schema of the constraint is consistent with what the table supports.
+     *
+     * @param constraint  The constraint being tested
+     *
+     * @throws IllegalArgumentException If there are columns referenced by the constraint unavailable in the table
+     */
+    /**
+     * Ensure that the schema of the constraint is consistent with what the table supports.
+     *
+     * @param constraint  The constraint being tested
+     *
+     * @throws IllegalArgumentException If there are columns referenced by the constraint unavailable in the table
+     */
+    private void validateConstraintSchema(DataSourceConstraint constraint) throws IllegalArgumentException {
+        Set<String> tableColumnNames = getSchema().getColumnNames();
+        // Validate that the requested columns are answerable by the current table
+        if (!constraint.getAllColumnNames().stream().allMatch(tableColumnNames::contains)) {
+            String message = String.format(
+                    "Received invalid request requesting for columns: %s that is not available in this table: %s",
+                    constraint.getAllColumnNames().stream()
+                            .filter(name -> !tableColumnNames.contains(name))
```

Comment:
I don't think so. Method references cannot be used in expressions like that (as far as I know)